### PR TITLE
fix(ios): undefined is not an object (evaluating 'measureSpec.setColumn')

### PR DESCRIPTION
**Pull request by m-abs**
We, the rest of the NativeScript community, thank you for your
contribution! 
To help the rest of the community review your change, please follow the instructions in the template.
-->

## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [x] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- ~[ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md~

## What is the current behavior?

In the past month we've tracked 20+ crashes with this error:
```
TypeError: undefined is not an object (evaluating 'measureSpec.setColumn')
```

We haven't been able to consistently reproduce the crash locally, but I think it is clear where it does wrong.

## What is the new behavior?

`GridLayout(ios).onMeasure(...)` will create `MeasureSpecs` if missing the for `child` view when it calls `this.eachLayoutChild(...)`

Fixes: #7226

### CODE REVIEW

File: tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts [hunk 1]
```diff
@@ -136,6 +136,11 @@ export class GridLayout extends GridLayoutBase {
 
         this.eachLayoutChild((child, last) => {
             let measureSpecs = this.map.get(child);
+            if (!measureSpecs) {
```
Comment (manoldonev) [inline tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts]: I was reviewing our code and I am not sure trying to fix this exceptional situation by registering layout child here is a good idea. How about just skipping this specific layout child instead (as obviously something is not right with its current state)?
``` ts
if (!measureSpecs) { 
    continue;
}
```

Comment (m-abs) [reply tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts]: I've finally managed to reproduce the problem see: https://github.com/NativeScript/NativeScript/issues/7226#issuecomment-491634072

I'm not sure what the best solution is or if we just use components+layouts incorrectly.

Just skipping as you suggest results in the component not being rendered.

Comment (manoldonev) [reply tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts]: hey @m-abs thank you isolating the problem!
So just skipping the iteration when measureSpec is undefined is not an option, ok. Will investigate the code on the nativescript-angular side a bit but if I don't find a better solution I am going to merge yours as is.

Meanwhile, there is also a workaround that you can use -- wrap the contents of your sub1 component (sub1.component.html) in any kind of layout like this:
``` html
<GridLayout>
    <ns-sub2></ns-sub2>
</GridLayout>
```

Comment (manoldonev) [reply tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts]: @m-abs I think we found the root of the problem in the ProxyViewContainer class -- see my commit here: https://github.com/NativeScript/NativeScript/pull/7230/commits/3acfc70ed49d3eca19666a603cfc62d50489ba66

**Comment by manoldonev** [comment]
test

File: tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts (hunk 1 above)
Comment (m-abs) [reply tns-core-modules/ui/layouts/grid-layout/grid-layout.ios.ts]: Looks good to me.

**Pull request merged**